### PR TITLE
Clarify asset reserved and asset reservation in progress in activity UI

### PR DESCRIPTION
**Pull request by rstorey**
Display different messages for when asset reservation is in progress and when it cannot be obtained.

### CODE REVIEW

File: concordia/static/js/action-app/index.js [hunk 1]
```diff
@@ -828,9 +828,14 @@ export class ActionApp {
 
         let {canEdit, reason} = this.canEditAsset(asset);
 
-        if (!this.assetReserved) {
+        if (this.assetReservationInProgress) {
```
Comment (acdha) [inline concordia/static/js/action-app/index.js]: Couldn't this be inferred from `this.assetReservationURL` being defined?

Comment (rstorey) [reply concordia/static/js/action-app/index.js]: I don't think so, because the reservationURL stays alive as long as the viewer is open.

Comment (acdha) [reply concordia/static/js/action-app/index.js]: Maybe I'm rushing out the door, but wouldn't the sequence be something like this?

1. Open viewer
2. asset reservation URL is set if there isn't some hard policy reason blocking editing (e.g. self-review)
3. XHR happens for the reservation
4. “in progress” message displays
5. `assetReserved` is set true on success

It seemed like `assetReservationInProgress` is defined roughly the same as `this.assetReservationURL && !this.assetReserved` would be from the perspective of the UI badging.

Comment (rstorey) [reply concordia/static/js/action-app/index.js]: The asset reservation URL isn't deleted until the viewer is closed.

Comment (acdha) [reply concordia/static/js/action-app/index.js]: Isn't that what we want? If it's defined, we're at some stage in the process of attempting to get a reservation (either waiting on XHR or retrying an error) but the intent is to attempt to be able to work on it until the viewer is closed?

This is also getting improved in #964 to consolidate the different places which handle open asset checks. So far I haven't found a situation where that combination isn't enough.

Comment (rstorey) [reply concordia/static/js/action-app/index.js]: I look forward to seeing your solution on indicating to end users when the asset reservation is in progress and when it's either successful or failed.

**Pull request closed without merging**